File: app/src/main/java/fr/frogdevelopment/nihongo/dico/Sense.java
package fr.frogdevelopment.nihongo.dico;

import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;

import java.util.HashSet;
import java.util.Set;

public class Sense {

    public final Set<String> pos = new HashSet<>();
    public final Set<String> field = new HashSet<>();
    public final Set<String> misc = new HashSet<>();
    public final Set<String> dial = new HashSet<>();
    public final Set<String> gloss = new HashSet<>();

    @Override
    public String toString() {
        ToStringBuilder builder = new ToStringBuilder(this, ToStringStyle.SIMPLE_STYLE);
        if (!pos.isEmpty()) {
            builder.append("pos", pos);
        }

        if (!field.isEmpty()) {
            builder.append("field", field);
        }

        if (!misc.isEmpty()) {
            builder.append("misc", misc);
        }

        builder.append("gloss", gloss);

        return builder.toString();
    }
}

File: app/src/main/java/com/frogdevelopment/nihongo/dico/data/entities/Sense.kt
package com.frogdevelopment.nihongo.dico.data.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = "senses", indices = [Index(value = ["entry_seq"]), Index(value = ["sense_seq"], unique = true)])
class Sense(
        @PrimaryKey(autoGenerate = true) val rowid: Int,
        @ColumnInfo(name = "entry_seq") val entrySeq: String,
        @ColumnInfo(name = "sense_seq") val senseSeq: String,
        @ColumnInfo val pos: String?,
        @ColumnInfo val field: String?,
        @ColumnInfo val misc: String?,
        @ColumnInfo val info: String?,
        @ColumnInfo val dial: String?,
        @ColumnInfo val gloss: String
)